### PR TITLE
Refactor GA + add User Id

### PR DESCRIPTION
**Pull request by NickHeiner**
This is currently failing on Travis. I'm going to investigate later, but feel free to hold off looking until then.

Connect #2847.

This will require https://github.com/department-of-veterans-affairs/appeals-deployment/pull/741 to work.

I started off this PR just adding the [user ID](https://developers.google.com/analytics/devguides/collection/analyticsjs/field-reference#userId). That's a 3-line change. However, I was concerned that I was unable to verify that it was working locally. That led me to refactor our GA usage to always fire, instead of being limited to prod. This gives us the benefit of being able to inspect GA traffic locally. For instance:

image

(We can see that `uid = 8`, which is the value we're trying to set in this PR.)

To keep the GA dashboard meaningful, we'll have one GA property for prod, and one property for everything else.

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -67,30 +67,29 @@
...
-        ga('send', 'pageview', page);
-      }
+  <script>
+    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+          (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+    })(window,document,'script', '<%= Rails.configuration.google_analytics_host %>','ga');
+
+    // Allow both caseflow domains to send analytics data
+    ga('create', '<%= Rails.configuration.google_analytics_account %>', 'auto', {
```
Comment (NickHeiner) [inline app/views/layouts/application.html.erb]: What is the point of doing this? On non-http, it causes us to make a request to `http` GA first, then get redirected. 

image

Can we just go straight to `https`?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -67,30 +67,29 @@
...
+      userId: '<%= current_user.id %>'
```
Comment (NickHeiner) [inline app/views/layouts/application.html.erb]: This was the one line that this PR was initially for. Is this the right ID to use? It needs to be non-PII, but something that we can link to a specific user.

**Comment by NickHeiner** [comment]
This question got swallowed by GitHub but still applies: https://github.com/department-of-veterans-affairs/caseflow/pull/2852#discussion_r131241325.

**Comment by NickHeiner** [comment]
Oops – didn't mean to close. :smile:

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (NickHeiner) [inline app/views/layouts/application.html.erb]: Talked to @shanear, and this is resolved. Also, I was commenting on the wrong line. :smile:

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -67,30 +67,29 @@
...
-      window.analyticsPageView = function(page) {
-        ga('send', 'pageview', page);
-      }
+  <script>
+    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+          (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+        m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+    })(window,document,'script', 'https://www.google-analytics.com/analytics.js', 'ga');
+
+    // Allow both caseflow domains to send analytics data
```
Comment (shanear) [inline app/views/layouts/application.html.erb]: we can delete this comment

**Pull request merged**